manim/mobject/mobject.py

```diff
@@ -383,15 +383,15 @@ def add_to_back(self, *mobjects: "Mobject") -> "Mobject":
         :meth:`add`
 
         """
+        if self in mobjects:
+            raise ValueError("Mobject cannot contain self")
+
         for mobject in mobjects:
-            if self in mobjects:
-                raise ValueError("Mobject cannot contain self")
             if not isinstance(mobject, Mobject):
                 raise TypeError("All submobjects must be of type Mobject")
 
-        filtered = list_update(mobjects, self.submobjects)
         self.remove(*mobjects)
-        self.submobjects = list(filtered) + self.submobjects
+        self.submobjects = list(set(mobjects)) + self.submobjects
```

Comment:
will this change something in the time performance of manim in scenes with lots of mobjects?

Reply:
shouldn't this destroy the order of mobjects?

Reply:
Converting between set and list should be slightly faster than using `list_update`, which uses `for...in` operation to iterate through the entire list.

Running this example shows these results:
```py
import time

l = {1, 1, 2, 2, 3, 3, 4, 4, 5, 5}

def list_update(l1, l2):
    """
    Used instead of list(set(l1).update(l2)) to maintain order,
    making sure duplicates are removed from l1, not l2.
    """
    return [e for e in l1 if e not in l2] + list(l2)

t0 = time.time()
l1 = list(set(l))
t1 = time.time()

t2 = time.time()
l2 = list_update(l, [])
t3 = time.time()

total = t1-t0
total2 = t3-t2
print(total)
print(total2)
```
```
6.9141387939453125e-06
8.821487426757812e-06


** Process exited - Return Code: 0 **
Press Enter to exit terminal
```

Reply:
> shouldn't this destroy the order of mobjects?

Oh right, the ordering. I will look into that.

Reply:
Thanks a lot for your timing test, this is definitely fast enough for a small amount of data

Reply:
> Oh right, the ordering. I will look into that.

This is important than the speed in this case.

Reply:
Comparing the following timings, it seems like the new changes are slower on average, but it does solve the duplication and ordering issue.
```py
l1 = list(dict.fromkeys(l))
l2 = list_update(l, [])
```
```
1.3828277587890625e-05
1.0728836059570312e-05
-------------------------------
8.344650268554688e-06
7.152557373046875e-06
-------------------------------
1.2874603271484375e-05
1.6450881958007812e-05
```